Update 01-upgrading-to-v1.6.md (#3386)
fishtown-analytics/docs.getdbt.com: website/docs/guides/migration/versions/01-upgrading-to-v1.6.md

### Behavior changes

:::info Action required if your project defines `metrics`

The [spec for metrics](https://github.com/dbt-labs/dbt-core/discussions/7456) has changed and now uses [MetricFlow](/docs/build/about-metricflow). 

:::

> If your dbt project defines metrics, you'll need to migrate to dbt v1.6 because the YAML spec has moved from dbt_metrics to MetricFlow. This means any tests you have won't compile on v1.5 or older. 

Reviewer: ```suggestion
If your dbt project defines metrics, you must migrate to dbt v1.6 because the YAML spec has moved from dbt_metrics to MetricFlow. Any tests you have won't compile on v1.5 or older. 
```

### Behavior changes

> :::info Action required if your project defines metrics 

Author: #3503

> :::info Action required if your project defines metrics 

Author: ```suggestion
:::info Action required if your project defines `metrics`
```

The [spec for metrics](https://github.com/dbt-labs/dbt-core/discussions/7456) has changed and now uses [MetricFlow](/docs/build/about-metricflow). 

:::

> If your dbt project defines metrics, you'll need to migrate to dbt v1.6 because the YAML spec has moved from dbt_metrics to the MetricFlow. This means any tests you have won't compile on v1.5 or older. 

Reviewer 2: Suggestion to remove extraneous word:
```suggestion
If your dbt project defines metrics, you'll need to migrate to dbt v1.6 because the YAML spec has moved from dbt_metrics to MetricFlow. This means any tests you have won't compile on v1.5 or older. 
```

## New and changed documentation

> - [Build your Semantic Layer](adding link) - Define your metrics with semantic models and MetricFlow using the command line (CLI) for dbt Core v1.6 beta. 

Author: ```suggestion
- [Build your Semantic Layer](/docs/build/build-semantic-layer-intro) - Define your metrics with semantic models and MetricFlow using the command line (CLI) for dbt Core v1.6 beta. 
```

> - [Build your Semantic Layer](/docs/build/build-semantic-layer-intro) - Define your metrics with semantic models and MetricFlow using the command line (CLI) for dbt Core v1.6 beta. 

Author: ```suggestion
- [Build your metrics](/docs/build/build-metrics-intro) - Define your metrics with semantic models and MetricFlow, a key component of the dbt Semantic Layer. Available on the command line (CLI) for dbt Core v1.6 beta or higher.
```
